### PR TITLE
DEPR: Turn ambiguous fields warnings into errors

### LINKED ISSUE #3235: Make sure the docs use tuple field accesses
Following #3101 and #3194, ambiguous field accesses have been deprecated. We however have still many examples in the doc which use the old ambiguous syntax.

We should (gradually) update the doc to make sure no field accesses are ambiguous.

### LINKED ISSUE #3416: DEPR: raise an error in case of ambiguous fields
### Bug report

**Bug summary**

Ambiguous fields were deprecated in yt 4.0 and targeted for "removal" in 4.1, see
https://github.com/yt-project/yt/blob/d10b207f7a660de2736e74f83b4c4c94bca1a0e2/yt/data_objects/static_output.py#L821

I am intentionally leaving this out of the otherwise comprehensive deprecation cycle PR #3240 because it's the only non-trivial bit to fix, but I'm opening this issue as a reminder to deal with this before yt 4.1

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary
Goals:
- close #3416
- fix corresponding errors in our documentation (if any) -> close #3235 (edit: no errors to report, issue closed)

this depends on #3948 (edit: now merged)

### CODE REVIEW

File: yt/fields/tests/test_ambiguous_fields.py [hunk 1]
```diff
@@ -16,7 +15,7 @@ def _mock_field(field, data):
     ds.add_field(("gas", "mock_field"), _mock_field, sampling_type="cell")
 
     # Test warnings are issued for ambiguous fields
```
Comment (neutrinoceros) [inline yt/fields/tests/test_ambiguous_fields.py:17]: I should edit this line too

**Comment by neutrinoceros** [comment]
Rebased. I'll make sure this doesn't break documentation before I open this for review.

**Comment by neutrinoceros** [comment]
rebased again to resolve a merge conflict

**Comment by neutrinoceros** [comment]
Could @matthewturk and/or @cphyc take a look ?

**Review by Xarthisius — APPROVED** [review]

**Review by jzuhone — APPROVED** [review]

**Pull request merged**